pkg/controller/cassandradatacenter/controller/configmap.go

```diff
@@ -12,48 +12,31 @@ import (
 	"strings"
 )
 
-// TODO: better name? This "conflicts" with corev1.VolumeMount
-type VolumeMount struct {
-	Volume    corev1.Volume
-	MountPath string
-}
+const (
+	MEBIBYTE = 1 << 20
+	GIBIBYTE = 1 << 30
+)
 
-type VolumeMounts []*VolumeMount
 
-type configurationResources struct {
-	cdc              *cassandraoperatorv1alpha1.CassandraDataCenter
-	configMap        *corev1.ConfigMap
-	volumeSource     *corev1.ConfigMapVolumeSource
-	seedNodesService *corev1.Service
-}
+func createOrUpdateOperatorConfigMap(rctx *reconciliationRequestContext, seedNodesService *corev1.Service) (*corev1.Volume, error) {
+	configMap := &corev1.ConfigMap{ObjectMeta: DataCenterResourceMetadata(rctx.cdc, "operator-config")}
 
-func CreateOrUpdateOperatorConfigMap(reconciler *CassandraDataCenterReconciler, cdc *cassandraoperatorv1alpha1.CassandraDataCenter, seedNodesService *corev1.Service) (*VolumeMount, error) {
-
-	configMap := &corev1.ConfigMap{ObjectMeta: DataCenterResourceMetadata(cdc, "operator-config")}
 	volumeSource := &corev1.ConfigMapVolumeSource{LocalObjectReference: corev1.LocalObjectReference{Name: configMap.Name}}
 
-	config := configurationResources{
-		cdc:              cdc,
-		configMap:        configMap,
-		volumeSource:     volumeSource,
-		seedNodesService: seedNodesService,
-	}
-
-	_, err := controllerutil.CreateOrUpdate(context.TODO(), reconciler.client, config.configMap, func(_ runtime.Object) error {
-
-		if err := addCassandraYamlOverrides(config); err != nil {
-			return err
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), rctx.client, configMap, func(_ runtime.Object) error {
+		addFileFn := func (path string, data string) {
```

Comment:
Each `addXYZ` function needs to modify both the `ConfigMap` and `ConfigMapVolumeSource` objects to add a "file". Previously we were passing around a `configurationResources` "context" object that contained these as fields so that each `addXYZ` function could call `configMapVolumeAddTextFile(...)`, but I feel that becomes a "god struct" -- better to break it up so that each function only takes its direct dependencies.

This IMO is cleaner, but I had two options:
- this approach, with a function parameter/reference/lambda (what's the Go term?) being passed around
- a struct that contains both the `ConfigMap` and `ConfigMapVolumeSource`, with a `addFile()` method receiver

Reply:
I personally like this approach. It's quite clean and very common in Go.

Also, I read a bit on receiver functions vs general functions, and it seems that you use receiver function when you either implement an interface, group methods or changing the state of the struct (that is the receiver itself). In other cases you use just functions.

Reply:
> changing the state of the struct

Which sort-of applies in this case. E.g., say we had a struct `ConfigMapArchive`, that had both the CM and CMVS as fields, then `configMapArchive.addFile(<path>, <data>)` makes sense too, right?

Reply:
yes, that would make sense.

I still like this current approach better.

Reply:
+1 on narrowing down what we just need to avoid god structs but it requires discipline not to make it god struct again as it might seem tempting to "just pass this little bit into that too" :)

Reply:
i would go with this approach as is

Reply:
I am just thinking how we would unit test the functions we are passing this addFileFn into. While calling it, I would need to make my own implementation of addFileFn?

Reply:
Yes, but that means its easy to mock out into a no-op, or something that checks the result easily...